### ISSUE TITLE
NullPointerException when Mapping object with collection to existing object with collection

### ISSUE THREAD

**Issue by tomorrow10** [modelmapper/modelmapper#249]
I have a problem mapping object with collection to another existing object with collection.

I have a pojo Agent class which has property of type Set<String> that I want to map to domain Agent class with property Set<Role>. I have a converter for String -> Role. When i call `modelMaper.map(pojoAgent, DomainAgent.class)`, everything works fine.

However, when I call `modelMaper.map(pojoAgent, domainAgent)`, I get a NullPointerException.

```
org.modelmapper.MappingException: ModelMapper mapping errors:

1) Converter org.modelmapper.internal.converter.CollectionConverter@60ee7a51 failed to convert java.util.Set to java.util.Set.

1 error

	at org.modelmapper.internal.Errors.throwMappingExceptionIfErrorsExist(Errors.java:374)
	at org.modelmapper.internal.MappingEngineImpl.map(MappingEngineImpl.java:69)
	at org.modelmapper.ModelMapper.mapInternal(ModelMapper.java:528)
	at org.modelmapper.ModelMapper.map(ModelMapper.java:412)
	at sk.slsp.omnichannel.usermanagement.rest.controller.AgentController.synchronizeAgent(AgentController.java:44)
	at sk.slsp.omnichannel.usermanagement.rest.controller.AgentControllerTest.testSomething(AgentControllerTest.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	at org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:51)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:237)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.lang.NullPointerException
	at org.modelmapper.internal.MappingEngineImpl.destinationProperty(MappingEngineImpl.java:343)
	at org.modelmapper.internal.MappingEngineImpl.typeMap(MappingEngineImpl.java:114)
	at org.modelmapper.internal.MappingEngineImpl.map(MappingEngineImpl.java:92)
	at org.modelmapper.internal.converter.IterableConverter.convert(IterableConverter.java:49)
	at org.modelmapper.internal.MappingEngineImpl.convert(MappingEngineImpl.java:366)
	at org.modelmapper.internal.MappingEngineImpl.map(MappingEngineImpl.java:96)
	at org.modelmapper.internal.MappingEngineImpl.setDestinationValue(MappingEngineImpl.java:252)
	at org.modelmapper.internal.MappingEngineImpl.propertyMap(MappingEngineImpl.java:186)
	at org.modelmapper.internal.MappingEngineImpl.typeMap(MappingEngineImpl.java:135)
	at org.modelmapper.internal.MappingEngineImpl.map(MappingEngineImpl.java:92)
	at org.modelmapper.internal.MappingEngineImpl.map(MappingEngineImpl.java:60)
	... 33 more

```

Is such mapping supported? Or do I have to set the mapping for the roles property to null and then handle it manually? I am using model-mapper 1.0.0

**Comment by chhsiao90** [comment]
@tomorrow10 - Thanks for the report. The exception stacktrace looks really weird, I will take a look!

**Comment by chhsiao90** [comment]
@tomorrow10 - This had been fixed by v1.1.0, please see that it fixed your problem, thanks!

**Comment by tomorrow10** [comment]
@chhsiao90 hi, i tested it and it seems to b working correctly now
